EMS/DAL/IEmployeeDal.cs

```diff
@@ -0,0 +1,16 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace EmployeeManagement;
+public interface IEmployeeDal
+{
+    List<Employee> FetchData(EmployeeFilter? employeeFilterInput);
+
+    List<Employee> FetchData();
+
+    bool Insert(List<Employee> data);
```

Comment:
insert and update methods doesn't have to be list of employees. just one employee. Also keep the parameter name as employee